qiita_db/data.py

```diff
@@ -615,6 +615,68 @@ def remove_filepath(self, fp):
         # Delete the files, if they are not used anywhere
         purge_filepaths(conn_handler)
 
+    def status(self, study):
+        """The status of the raw data within the given study
+
+        Parameters
+        ----------
+        study : Study
+            The study that is looking to the raw data status
+
+        Returns
+        -------
+        str
+            The status of the raw data
+
+        Raises
+        ------
+
+        Notes
+        -----
+        The status of the raw data is inferred by the status of the processed
+        data generated from this raw data. Since the raw data can be shared by
+        multiple studies; the raw data can have multiple status, so the status
+        is defined by the view from a study. If not processed data was
+        generated on the given study from this raw data; then the status is
+        'sandbox'.
+        """
+        if self._id not in study.raw_data():
+            raise QiitaDBStatusError(
+                "The study %s do not have access to the raw data %s"
+                % (study.id, self.id))
+
+        conn_handler = SQLConnectionHandler()
+        sql = """SELECT processed_data_status
+                FROM qiita.processed_data_status pds
+                  JOIN qiita.processed_data pd
+                    ON pds.processed_data_status_id=pd.processed_data_status_id
+                  JOIN qiita.preprocessed_processed_data ppd_pd
+                    ON ppd_pd.processed_data_id=pd.processed_data_id
+                  JOIN qiita.prep_template_preprocessed_data pt_ppd
+                    ON pt_ppd.preprocessed_data_id=ppd_pd.preprocessed_data_id
+                  JOIN qiita.prep_template pt
```

Comment:
Why the prep template joins in here? If we just need to get up to the processed data, using raw_data, study_raw_data, and study_preprocessed_data would be easier and faster than prep template.